frontend/src/libs/packages/middleware/libs/helpers/update-geolocation.helper.ts

```diff
@@ -0,0 +1,66 @@
+import { Config } from '~/libs/packages/config/config.package.js';
+import { type StartWatchTruckLocation } from '~/slices/trucks/types/start-watch-truck-location.type.js';
+
+import { notification } from '../../../notification/notification.js';
+import { ClientToServerEvent, socket } from '../../../socket/socket.js';
+import { GeolocationMessages } from '../enums/geolocation-messages.enum.js';
+import { type GeolocationLatLng } from '../types/types.js';
+
+type GetLocationError = (error: string) => void;
+
+type GetLocationSuccess = (location: GeolocationLatLng) => void;
+
+const socketInstance = socket.getInstance();
+
+const VITE_DRIVER_GEOLOCATION_UPDATE_INTERVAL = new Config().ENV.API
+  .VITE_DRIVER_GEOLOCATION_UPDATE_INTERVAL;
+
+let getLocationSuccess: GetLocationSuccess;
+let getLocationError: GetLocationError;
+const location = new Promise<GeolocationLatLng>((resolve, reject) => {
+  getLocationSuccess = resolve;
+  getLocationError = reject;
+});
+const handleUpdateSuccess = (position: GeolocationPosition): void => {
+  getLocationSuccess({
+    lat: position.coords.latitude,
+    lng: position.coords.longitude,
+  });
+};
+const handleUpdateError = (error: GeolocationPositionError): void => {
+  getLocationError(error.message);
+};
+
+const updateGeolocation = async (
+  payload: StartWatchTruckLocation,
+  setUpdateLocationInterval: (interval: NodeJS.Timer) => void,
+): Promise<boolean | null> => {
+  if (socketInstance) {
```

Comment:
Too big if body. It's better to 
```suggestion
  if (!socketInstance) { return null }
```